DOCS: Add documentation for StreamField group option (#13201)
wagtail/wagtail: docs/topics/streamfield.md

For a list of icons available out of the box, see our [icons overview](icons). Project-specific icons are also displayed in the [styleguide](styleguide).

### Grouping Blocks

> When a ``StreamField`` has a large number of different block types, the block picker menu can become long and difficult to scan. To help with this, you can organize related blocks into groups by passing the ``group`` option as a keyword argument or as an attribute on a ``Meta`` class. If no ``group`` is specified, a block will appear under the default "Common" heading.
> 
> Any blocks that share the same ``group`` name will be clustered together under that name as a heading in the menu, making the interface cleaner and more efficient for content authors.

Reviewer: ```suggestion
When a `StreamField` has a large number of different block types, the block picker menu can become long and difficult to scan. To help with this, you can organize related blocks into groups by passing the `group` option as a keyword argument or as an attribute on a `Meta` class. Grouped block types will show after all of the ungrouped (common) types, those with no `group` specified.

Any blocks that share the same `group` name will be clustered together under that name as a heading in the menu, making the interface cleaner and more efficient for content authors.
```

Removing backticks but a proposed re-word of the second sentence to call out ordering and also there is no "Common" heading, it's just the ungrouped section that goes first.
